From 1efb7b36f0afeb4c8236e6fb4fe7fbf87f4dd7e6 Mon Sep 17 00:00:00 2001
From: qiongz <qiongz@codeaurora.org>
Date: Wed, 14 Jun 2017 15:42:56 +0800
Subject: [PATCH 7/9] Fix imsi info error when insert some china operator card.

Imsi only show mccmnc when the phone type is cdma phone since the
length of return from RIL is 0 but not NULL for mMin. if mMin is
NULL or length is 0, mPhone.getSubscriberId() will get NULL when
the phone type is cdma, hence the flow of get imsi will move on
to retrieve the imsi from IccRecords.

Change-Id: Idda892c736db046d8924bb45a5fbb2a5f2a6d248
CRs-Fixed: 2212375
---
 .../com/android/internal/telephony/ServiceStateTracker.java     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/ServiceStateTracker.java b/src/java/com/android/internal/telephony/ServiceStateTracker.java
index 91f2cdea3..39940f033 100755
--- a/src/java/com/android/internal/telephony/ServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/ServiceStateTracker.java
@@ -1745,7 +1745,7 @@ public class ServiceStateTracker extends Handler {
                 .getSystemService(Context.TELEPHONY_SERVICE))
                 .getSimOperatorNumericForPhone(mPhone.getPhoneId());
 
-        if (!TextUtils.isEmpty(operatorNumeric) && getCdmaMin() != null) {
+        if (!TextUtils.isEmpty(operatorNumeric) && !TextUtils.isEmpty(getCdmaMin())) {
             return (operatorNumeric + getCdmaMin());
         } else {
             return null;
-- 
2.28.0

